group-coordinator/src/test/java/org/apache/kafka/coordinator/group/assignor/RangeAssignorTest.java

```diff
@@ -0,0 +1,523 @@
+        expectedAssignment.put(consumerB, mkAssignment(mkTopicAssignment(topic1Uuid, 1)));
+        expectedAssignment.put(consumerC, mkAssignment(mkTopicAssignment(topic1Uuid, 2)));
+        // Topic 3 Partitions Assignment
+        expectedAssignment.get(consumerA).putAll(mkAssignment(mkTopicAssignment(topic3Uuid, 0)));
+        expectedAssignment.get(consumerB).putAll(mkAssignment(mkTopicAssignment(topic3Uuid, 1)));
+
+        assertAssignment(expectedAssignment, computedAssignment);
+    }
+
+    @Test
+    public void testReassignmentNumConsumersGreaterThanNumPartitionsWhenOneConsumerAdded() {
+        Map<Uuid, AssignmentTopicMetadata> topics = new HashMap<>();
+        topics.put(topic1Uuid, new AssignmentTopicMetadata(2));
+        topics.put(topic2Uuid, new AssignmentTopicMetadata(2));
+
+        Map<String, AssignmentMemberSpec> members = new TreeMap<>();
+
+        Map<Uuid, Set<Integer>> currentAssignmentForA = new HashMap<>();
+        currentAssignmentForA.put(topic1Uuid, Collections.singleton(0));
+        currentAssignmentForA.put(topic2Uuid, Collections.singleton(0));
+        members.put(consumerA, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Arrays.asList(topic1Uuid, topic2Uuid),
+            currentAssignmentForA
+        ));
+
+        Map<Uuid, Set<Integer>> currentAssignmentForB = new HashMap<>();
+        currentAssignmentForB.put(topic1Uuid, Collections.singleton(1));
+        currentAssignmentForB.put(topic2Uuid, Collections.singleton(1));
+        members.put(consumerB, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Arrays.asList(topic1Uuid, topic2Uuid),
+            currentAssignmentForB
+        ));
+
+        // Add a new consumer to trigger a re-assignment
+        members.put(consumerC, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Arrays.asList(topic1Uuid, topic2Uuid),
+            Collections.emptyMap()
+        ));
+
+        AssignmentSpec assignmentSpec = new AssignmentSpec(members, topics);
+        GroupAssignment computedAssignment = assignor.assign(assignmentSpec);
+
+        Map<String, Map<Uuid, Set<Integer>>> expectedAssignment = new HashMap<>();
+        // Topic 1 Partitions Assignment
+        expectedAssignment.put(consumerA, mkAssignment(mkTopicAssignment(topic1Uuid, 0)));
+        expectedAssignment.put(consumerB, mkAssignment(mkTopicAssignment(topic1Uuid, 1)));
+        // Topic 2 Partitions Assignment
+        expectedAssignment.get(consumerA).putAll(mkAssignment(mkTopicAssignment(topic2Uuid, 0)));
+        expectedAssignment.get(consumerB).putAll(mkAssignment(mkTopicAssignment(topic2Uuid, 1)));
+
+        // Consumer C shouldn't get any assignment, due to stickiness A, B retain their assignments
+        assertNull(computedAssignment.members().get(consumerC));
```

Comment:
This is interesting. Should we still create a member in this case but with an empty assignment?

Reply:
Yeah, that's right. It does not matter from the TargetAssignmentBuilder perspective. We can keep it as it is.